### ISSUE TITLE
Why is my reproduction result higher than the one in the paper? 

### ISSUE THREAD

**Issue by torotocat** [kinredon/Harmonious-Teacher#21]
1. class：|  person  |  rider   |  car       |  truck    |  bus   |  train  |  motorcycle  |  bicycle  |
2. my:     |  58.419  | 60.035  | 70.148 | 45.800 | 62.820 | 47.403  |    48.602    |  59.398   |
3. paper: |52.1       |55.8       |67.5      |   32.7    | 55.9     |   49.1   |    40.1         |  50.3       |

What is the cause of this issue? My Cityscapes and Foggy datasets were converted to VOC format using your script. For experimental rigor, I also used a different script to generate the VOC dataset, but the result is the same. Additionally, is the beta index for the Foggy dataset simply 0.02, or is there something else?

When reproducing the results, the only changes I made were updating a few packages due to conflicts; the rest of the environment setup is the same as in the paper.

**Comment by kinredon** [comment]
Yes, the foggy level is 0.02. Although the code in this repo may produce better results such as ~51.0, your results are abnormal and much higher than expectation. Based on the information provided by you, I cannot figure out what reason causing this issue. Do you directly train the model on the foggy cityscapes with labeled data?

**Comment by RyeChianti** [comment]
Can you reproduce this code normally? I downloaded the Dataset in the format requested by the dataset section and found that this problem kept coming up. FileNotFoundError: [Errno 2] No such file or directory: '/home/sysu/maindesktop/Harmonious-Teacher-main/cvpods/data/cityscape_multi_gpa/ImageSets/Main/trainval.txt'  But I did not find this file trainval.txt when I downloaded the data set. Could you please provide a method?

> 1. class：|  person  |  rider   |  car       |  truck    |  bus   |  train  |  motorcycle  |  bicycle  |
> 
>     2. my:     |  58.419  | 60.035  | 70.148 | 45.800 | 62.820 | 47.403  |    48.602    |  59.398   |
> 
>     3. paper: |52.1       |55.8       |67.5      |   32.7    | 55.9     |   49.1   |    40.1         |  50.3       |
> 
> 
> What is the cause of this issue? My Cityscapes and Foggy datasets were converted to VOC format using your script. For experimental rigor, I also used a different script to generate the VOC dataset, but the result is the same. Additionally, is the beta index for the Foggy dataset simply 0.02, or is there something else?
> 
> When reproducing the results, the only changes I made were updating a few packages due to conflicts; the rest of the environment setup is the same as in the paper.



> Yes, the foggy level is 0.02. Although the code in this repo may produce better results such as ~51.0, your results are abnormal and much higher than expectation. Based on the information provided by you, I cannot figure out what reason causing this issue. Do you directly train the model on the foggy cityscapes with labeled data?

**Comment by kinredon** [comment]
@RyeChianti You may need to follow and understand the VOC data structure. The trainval.txt is a text file where each line is the image filename without suffixes.

**Comment by RyeChianti** [comment]
> 1. class：|  person  |  rider   |  car       |  truck    |  bus   |  train  |  motorcycle  |  bicycle  |
> 
>     2. my:     |  58.419  | 60.035  | 70.148 | 45.800 | 62.820 | 47.403  |    48.602    |  59.398   |
> 
>     3. paper: |52.1       |55.8       |67.5      |   32.7    | 55.9     |   49.1   |    40.1         |  50.3       |
> 
> 
> What is the cause of this issue? My Cityscapes and Foggy datasets were converted to VOC format using your script. For experimental rigor, I also used a different script to generate the VOC dataset, but the result is the same. Additionally, is the beta index for the Foggy dataset simply 0.02, or is there something else?
> 
> When reproducing the results, the only changes I made were updating a few packages due to conflicts; the rest of the environment setup is the same as in the paper.

**Comment by RyeChianti** [comment]
> 1. class：|  person  |  rider   |  car       |  truck    |  bus   |  train  |  motorcycle  |  bicycle  |
> 
>     2. my:     |  58.419  | 60.035  | 70.148 | 45.800 | 62.820 | 47.403  |    48.602    |  59.398   |
> 
>     3. paper: |52.1       |55.8       |67.5      |   32.7    | 55.9     |   49.1   |    40.1         |  50.3       |
> 
> 
> What is the cause of this issue? My Cityscapes and Foggy datasets were converted to VOC format using your script. For experimental rigor, I also used a different script to generate the VOC dataset, but the result is the same. Additionally, is the beta index for the Foggy dataset simply 0.02, or is there something else?
> 
> When reproducing the results, the only changes I made were updating a few packages due to conflicts; the rest of the environment setup is the same as in the paper.

What machine configuration and code configuration did you reproduce, and why did I find that every item was lower

**Comment by RyeChianti** [comment]
We tried to reproduce your results, but found the results were a little different. Could you give us some guidance. To fit our machine we made the following modifications.In Harmonious_Teacher-main/ht_c2f/config.py NUM_WORKERS=4  -->NUM_WORKERS=1 ; BASE_LR=0.016 --> BASE_LR=0.004 ;IMS_PER_BATCH=16  --> IMS_PER_BATCH=4;
In Harmonious_Teacher-main/cvpods/dvpods/engine/setup.py, assert base_world_size == 8 --> assert base_world_size == 2


    class：| person | rider | car | truck | bus | train | motorcycle | bicycle |

        my: | 51.017 | 52.579 | 64.678 | 32.110 | 48.242 | 10.200 | 36.264 | 48.779 |

        paper: |52.1 |55.8 |67.5 | 32.7 | 55.9 | 49.1 | 40.1 | 50.3 |

In particular, the gap in train is very large, could you give us some guidance? We have repeated many times and found that the results are not good.

**Comment by RyeChianti** [comment]
@kinredon @torotocat We tried to reproduce your results, but found the results were a little different. Could you give us some guidance. To fit our machine we made the following modifications.In Harmonious_Teacher-main/ht_c2f/config.py NUM_WORKERS=4 -->NUM_WORKERS=1 ; BASE_LR=0.016 --> BASE_LR=0.004 ;IMS_PER_BATCH=16 --> IMS_PER_BATCH=4;
In Harmonious_Teacher-main/cvpods/dvpods/engine/setup.py, assert base_world_size == 8 --> assert base_world_size == 2

class：| person | rider | car | truck | bus | train | motorcycle | bicycle |

    my: | 51.017 | 52.579 | 64.678 | 32.110 | 48.242 | 10.200 | 36.264 | 48.779 |

    paper: |52.1 |55.8 |67.5 | 32.7 | 55.9 | 49.1 | 40.1 | 50.3 |

In particular, the gap in train is very large, could you give us some guidance? We have repeated many times and found that the results are not good.

**Comment by kinredon** [comment]
@RyeChianti How many GPUs do you use? The default configuration is tailored for the 4 GPUs.

**Comment by RyeChianti** [comment]
> [@RyeChianti](https://github.com/RyeChianti) How many GPUs do you use? The default configuration is tailored for the 4 GPUs.

1 GPU

**Comment by ohppy** [comment]
In the results of my reproduction, the performance of other categories and metrics such as AP50 are quite close to those reported in the original paper, but the performance of the "train" category is significantly lower. Could this be due to using a single GPU while retaining the default configuration? For single-GPU training, which parameters need to be fine-tuned and how?
